File: BE/src/main/java/com/haivn/common_api/ChucVu.java
package com.haivn.common_api;

import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.DynamicUpdate;
import org.hibernate.annotations.Where;

import javax.persistence.*;

@Entity
@Table(name = "chuc_vu")
@Getter
@Setter
@DynamicUpdate
@Where(clause = "deleted=false")
public class ChucVu extends BaseEntity{
    @Column(name = "name")
    private String name;
    @Column(name = "level")
    private Short level;
    @Column(name = "status")
    private Short status;

}

File: FE/lib/model/model_qltvn/chuc_vu.dart
class ChucVu {
  int? id;
  String? name;
  int? level;
  int? status;

  ChucVu({this.id, this.name, this.level, this.status});

  ChucVu.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        name = json['name'],
        level = json['level'],
        status = json['status'];

  Map<String, dynamic> toJson() => {'name': name, 'level': level,'status': status};
}
